sale_commission/tests/test_sale_commission.py

```diff
@@ -171,6 +171,7 @@ def _invoice_sale_order(self, sale_order, date=None):
         if date:
             invoice.invoice_date = date
             invoice.date = date
+        invoice.flush()
```

Comment:
Why do you need these `flush` if no SQL operation is done?

Reply:
We need to execute all compute. commission_amount is not executed until it is needed, but it is needed once it has been settled. Then an error is raised. In order to avoid this, we need to flush.

Reply:
Then put such explanation as comment here for the rest and for your future you for avoiding an accidental removal

Reply:
:wink:

Reply:
Thanks. Still something bugs me, as a correct dependency chain should trigger the recomputation, but I'm not going around more.